### PR TITLE
cherrypick from @spxtr #5835 Add support for something like PodPresets directly into prow.

### PR DESCRIPTION
**Pull request by BenTheElder**
This PR picks the first commit from https://github.com/kubernetes/test-infra/pull/5835 without the broken bump commit.

The existing PR has been blocked for a while on discussion and/or various rebases. We want to get this in and all we needed is to drop the broken bump commit so I've picked the main commit over to here.

### CODE REVIEW

**Comment by BenTheElder** [comment]
will rebase to pickup upstream types after lunch :man_shrugging:

File: prow/config/jobs.go [hunk 1]
```diff
@@ -17,12 +17,62 @@ limitations under the License.
...
 	"time"
 
 	"k8s.io/api/core/v1"
 )
 
+// Preset is intended to match the k8s' PodPreset feature, and may be removed
+// if that feature goes beta.
+type Preset struct {
+	Labels       map[string]string  `json:"labels"`
+	Env          []kube.EnvVar      `json:"env"`
```
Comment (stevekuznetsov) [inline prow/config/jobs.go]: does this compile? wrong import

Comment (krzyzacy) [reply prow/config/jobs.go]: should updated to v1 now

Comment (BenTheElder) [reply prow/config/jobs.go]: it was updated in the second commit. other than rebasing I didn't want to modify joe's commit.

**Comment by BenTheElder** [comment]
/hold cancel
/shrug
We will need to bump all the things

File: prow/config/jobs.go (hunk 1 above)
Comment (kargakis) [reply prow/config/jobs.go]: We are using go aliases now so kube.EnvVar is v1.EnvVar

**Pull request merged**